File: RB101_Programming_Foundations/Small Probems/Easy7/Python/end_is_near.py
# Write a method that returns the next to last word in the String passed to it as an argument.

# Words are any sequence of non-blank characters.

# You may assume that the input String will always contain at least two words.

'''
Input: string

Output: second to last word in the string

Rules


Data structure / Algorithms
    split the string into list by space
    return the -2 position in the array


'''

def penultimate(string):
    return string.split()[-2]


print(penultimate('last word') == 'last')
print(penultimate('Launch School is great!') == 'is')

File: RB101_Programming_Foundations/small_problems/Easy7/end_is_near.rb
# def penultimate(string)
#   words = string.split
#   words[-2]
# end


# p penultimate('last word') == 'last'
# p penultimate('Launch School is great!') == 'is'


def middle(string)
  words = string.split
  middle_idx = words.size / 2

  if words.empty?
    ''
  elsif words.size.odd?
    words[middle_idx]
  else
    words[middle_idx - 1] + " " + words[middle_idx]
  end
end


p middle('')
p middle('last')
p middle('last word')
p middle('last word here')
p middle('last word here near')
p middle('last word here world ends')